applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing_integration.h

```diff
@@ -0,0 +1,394 @@
+// ==============================================================================
+//  KratosShapeOptimizationApplication
+//
+//  License:         BSD License
+//                   license: ShapeOptimizationApplication/license.txt
+//
+//  Main authors:    Baumgärtner Daniel, https://github.com/dbaumgaertner
+//
+// ==============================================================================
+
+#ifndef MAPPER_VERTEX_MORPHING_INTEGRATION_H
+#define MAPPER_VERTEX_MORPHING_INTEGRATION_H
+
+// ------------------------------------------------------------------------------
+// System includes
+// ------------------------------------------------------------------------------
+#include <iostream>
+#include <string>
+
+// ------------------------------------------------------------------------------
+// External includes
+// ------------------------------------------------------------------------------
+
+// ------------------------------------------------------------------------------
+// Project includes
+// ------------------------------------------------------------------------------
+#include "mapper_vertex_morphing.h"
+#include "processes/find_conditions_neighbours_process.h"
+#include "utilities/math_utils.h"
+
+// ==============================================================================
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+/** Detail class definition.
+*/
+
+class MapperVertexMorphingIntegration : public MapperVertexMorphing

```

Comment:
Personally, I find this name a bit misleading since in the "regular" approach we are also integrating. However, in a very pragmatic way. @armingeiser : Do you think it makes sense to change it maybe to something like "MapperVertexMorphingImprovedIntegration"? Of course, this implies also a change of the corresponding file names.